src/UFS2ARCO/ufsdataset.py

```diff
@@ -81,11 +72,23 @@ def default_open_dataset_kwargs(self):
         }
         return kw
 
-    def __init__(self, path_in, config_filename):
+    def __init__(self, path_in: Callable, config_filename: str) -> None:
         super(UFSDataset, self).__init__()
         name = self.__class__.__name__  # e.g., FV3Dataset, MOMDataset
 
-        self.path_in = path_in
+        # create and initialze instance variable for class attributes
+        self.path_out: str = ""
+        self.forecast_hours: List[int] = []
+        self.file_prefixes: List[str] = []
+
+        self.chunks_in: Dict = {}
+        self.chunks_out: Dict = {}
+        self.coords: List[str] = []
+        self.data_vars: List[str] = []
+        self.zarr_name: str = ""
+
+        self.path_in: Callable = path_in
+
```

Comment:
This fixes the potential issue that I was mentioning, looks good to me

Reply:
One issue though is that this will override the defaults set by FV3Dataset and so on. I'll make a suggestion further down in the code.